Sources/Resolver/Resolver.swift

```diff
@@ -453,6 +460,17 @@ public struct ResolverOptions<Service> {
 
     // MARK: - Fuctionality
 
+    @discardableResult
+    public func context(_ resolver: Resolver) -> ResolverOptions<Service> {
+        let exisitingFactory = registration.factory
+        registration.factory = { [weak resolver] r, a in
+            Resolver.tempContext = resolver
+            defer { Resolver.tempContext = nil }
```

Comment:
A single static "temp" and resetting to nil seems very wrong to me, since Resolver can resolve dependencies recursively. What happens when a dependent service instantiates a dependent service that also uses this technique?